src/components/Pressable/PressableWithFeedback.js

```diff
@@ -24,21 +25,41 @@ const PressableWithFeedbackDefaultProps = {
 
 const PressableWithFeedback = forwardRef((props, ref) => {
     const propsWithoutStyling = _.omit(props, omittedProps);
+    const [disabled, setDisabled] = useState(props.disabled);
+
+    useEffect(() => {
+        setDisabled(props.disabled);
+    }, [props.disabled]);
+
     return (
         <GenericPressable
             ref={ref}
             style={props.wrapperStyle}
             // eslint-disable-next-line react/jsx-props-no-spreading
             {...propsWithoutStyling}
+            disabled={disabled}
+            onPress={(e) => {
+                setDisabled(true);
+                const onPress = props.onPress(e);
+                InteractionManager.runAfterInteractions(() => {
+                    if (!(onPress instanceof Promise)) {
+                        setDisabled(props.disabled);
```

Comment:
The `props.disabled` that we are using inside the `InteractionManager.runAfterInteractions` callback or the one after the promise is resolved (few lines below) is evaluated at the time we call onPress and that value is used later. Meaning props.disabled does not reflect the current state but the state that we evaluated when onPress was initially called.

To better explain, here is an example:
1. props.disabled is false
2. Call onPress
3. Evaluate the promise callback, After the promise is resolved we will call setDisabled(false) // props.disabled is evaluated now
4. Promise is not resolved yet (button action still being executed)
5. props.disabled is set to true
6. Promise is resolved
7. Now we will call the previously evaluated callback which is setDisabled(false) even though the current props.disabled is true


This logic is copied from OptionRow but we missed that case in both PRs which lead to a regression https://github.com/Expensify/App/issues/20983.